task9/delivery/meteorController.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response, NextFunction, Router } from 'express'
+import { StatusCodes } from 'http-status-codes'
+import MeteorMapper from '../use_cases/MeteorMapper'
+import Exception from '../exception/Exception'
+import RoverMapper from '../use_cases/RoverMapper'
+import { imageRequestSchema, meteorQuerySchema } from '../validation/schemas'
+import Validator from '../validation/Validator'
+import { MeteorQuery } from '../models/meteorQuery'
+import { MeteorResponse } from '../models/meteor/meteorResponse'
+import { DateRange } from '../models/DateRange'
+import { ImageRequest } from '../models/imageRequest'
+
+class MeteorController {
+  private readonly router: Router
+  private meteorValidator: Validator
+  private imageValidator: Validator
+
+  constructor() {
+    this.router = Router()
+    this.meteorValidator = new Validator(meteorQuerySchema)
+    this.imageValidator = new Validator(imageRequestSchema)
+    this.initializeRoutes()
+  }
+
+  private initializeRoutes() {
+    this.router.get('/meteors', this.meteorValidator.validate, this.getMeteors)
+    this.router.get('/form', this.getForm)
+    this.router.post('/image', this.imageValidator.validate, this.getImage)
+  }
+
+  private getMeteors = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const meteorQuery: MeteorQuery = request.query
+      const dateRange: DateRange = DateRange.createFromDate(meteorQuery.date)
+      const meteorDto: MeteorResponse = await MeteorMapper.getMeteorDto(
+        dateRange.startDate,
+        dateRange.endDate,
+        meteorQuery.wereDangerousMeteors,
+        meteorQuery.count
+      )
+      response.render('meteors.njk', meteorDto)
+    } catch (error) {
+      if (error instanceof Exception) {
+        next(
+          new Exception(
+            error.code,
+            `Failed to get the meteor data due to: ${error.message}`
+          )
+        )
+      } else {
+        next(
+          new Exception(
+            StatusCodes.INTERNAL_SERVER_ERROR,
+            'An unknown error occurred'
+          )
+        )
+      }
```

Comment:
maybe better to leave only `next(error)` and then move this logic into errorHandler